### ISSUE TITLE
Year formatting question

### ISSUE THREAD

**Issue by hethcox** [malcommac/SwiftDate#616]
I'm writing an app using SwiftDate 5.0.4. I noticed something strange. This line of code:

```swift
print("2018-12-31".toDate()?.toFormat("MMM dd YYYY"))
```
produces the output: 

```swift
Optional("Dec 31 2019")
```

Is this a bug or am I misusing the API?

**Comment by hethcox** [comment]
This looks like a serious bug to me. Can someone verify the are/aren't seeing this behavior?

**Comment by mSauvestre** [comment]
Hey hethcox,

I think you are unaware of the full behavior of print(). If the passed element is an optional, print will let you know so by displaying this "Optional()" thing.

If you want your date displayed normally in this case, you would have to do:
if let aDate = "2018-12-31".toDate() {
    print(aDate.toFormat("MMM dd YYYY"))
}

Because aDate in this case isn't an optional anymore - in opposition to "2018-12-31".toDate(), which returns an object of type Date? (you can clearly see it to the ? before your function call toFormat) -, print won't be displaying this "Optional()" thingy :)

Hope that helps !

TLDR: It's normal, SwiftDate is fine, you aren't misusing the API, but you misunderstood a concept of print()

**Comment by hethcox** [comment]
Hi mSauvestre,
I'm not talking about how Swift outputs Optionals. What I'm showing that I created a SwiftDate for the last day of this year, but it printed out the last day of next year.

**Comment by mSauvestre** [comment]
Oh, sorry, you're completely right. I didn't see that element, my bad !

Definitively a bug. Seems to come from the formatter rather than the parser. When parsing the date, it's initialised correctly. When formatted, the information on the year gets corrupted.

Really related to the 31st of December also 👍

**Comment by Valpertui** [comment]
It's not a bug.  
It's the difference between YYYY and yyyy.  
YYYY represent the year of the current week (Week based calendar). The 31 december 2018 is a Sunday, which is also the first day of the first week of 2019 for week based calendars.
yyyy represent the year of the day (Day based calendar).
See https://www.unicode.org/reports/tr35/tr35-31/tr35-dates.html#Date_Format_Patterns for more informations

**Comment by mSauvestre** [comment]
Well spotted 🙏

**Comment by malcommac** [comment]
Thanks for answering
